zipline/src/hostMain/kotlin/app/cash/zipline/internal/topologicalSort.kt

```diff
@@ -59,7 +59,24 @@ internal fun <T> List<T>.topologicalSort(sourceToTarget: (T) -> Iterable<T>): Li
   }
 
   require(result.size == this.size) {
-    "No topological ordering is possible for $this"
+    buildString {
+      append("No topological ordering is possible for these items:")
+
+      val unorderedItems = toMutableSet()
+        .apply {
+          removeAll(result.toSet())
+        }
```

Comment:
Not that it matters, so feel free to ignore, but since we don't care about performance right here...
```suggestion
      val unorderedItems = toSet() - result.toSet()
```

Reply:
That’s what I did first, and for whatever reason it inferred the wrong type...

Reply:
Ahhh . ..  `buildString()` has a receiver that defines `toSet()`. Fix is a little ugly, but okay:
```
      val unorderedItems = this@topologicalSort.toSet() - result.toSet()
```

Reply:
I don't feel strongly. Leave it as is

Reply:
I prefer your recommendation!